### ISSUE TITLE
firefox on darwin pulls in pipewire, causes eval error

### ISSUE THREAD

**Issue by sheeeng** [NixOS/nixpkgs#366581]
## Describe the bug

```text
 error: Package ‘pipewire-1.2.6’ in /nix/store/6zgbbqlr7nnfxpzkyj7fsl4fpg89jbw0-source/pkgs/development/libraries/pipewire/default.nix:223 is not available on the requested hostPlatform:
hostPlatform.config = "aarch64-apple-darwin"
```

## Steps To Reproduce

```nix
{ pkgs, ... }:
{
  config = {
    programs.firefox = {
      enable = true; # https://nix-community.github.io/home-manager/options.xhtml#opt-programs.firefox.enable
    };
  };
}
```

## Expected behavior

The code should enable Firefox.

## Metadata

```console
$ nix-shell --packages nix-info --run "nix-info --markdown"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/9bdl6wa2qck3q0f7l892plk5lwy3pvx1-source`
```

## Notify maintainers

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by JohnRTitor** [comment]
CC @mweinelt

**Comment by Atemu** [comment]
Hold on, no I missed that https://github.com/NixOS/nixpkgs/pull/350384 added support for Firefox on Darwin.

cc @booxter

**Comment by Atemu** [comment]
Btw, please reframe your issues you take to Nixpkgs without depending on 3rd party projects. This has nothing to do with the home-manager module, it's a bug in the firefox derviation itself.

**Comment by booxter** [comment]
`firefox` the package (the wrapped version) is not - yet? - supported for Darwin. Please use `firefox-unwrapped`.

**Comment by ghenricc** [comment]
Pulling in `pipewire` can be avoided by simply wrapping `firefox-unwrapped` yourself, ie. 

```nix
{ pkgs ? import ./. {} }: with pkgs; (wrapFirefox.override {pipewire = null;}) firefox-unwrapped {}
```

However, `wrapFirefox` makes additional assumptions, ie. the location of the binary and other files, that do not hold up when considering Darwin devices. 

Although I don't expect a fix to this would be too difficult or take long enough to require it, should `firefox` be marked as broken on `x86_64-darwin` and `aarch_64-darwin` to help direct users the right direction?

**Comment by booxter** [comment]
@ghenricc You are right. It would be helpful, for the time being, to raise a legible error message directing to unwrapped flavor.

Eventually, we should make the wrapper support Darwin. I don't think the existing Linux wrapper cannot be reused as-is, for the simple fact that Darwin apps are bundled into .app directories and their binaries cannot be simply substituted with a script setting envvars. For example, variables would have to be injected through the package `plist`. I don't think it's as simple as adjusting some paths in the existing wrapper, sadly.

I will add this error message to my backlog, but it make take a while. If you are up to, feel free to send a PR for this. 🙏

**Comment by booxter** [comment]
OK I went ahead and sent this after all: https://github.com/NixOS/nixpkgs/pull/376849

**Comment by booxter** [comment]
FYI we had a discussion on the matter yesterday in Matrix and I stand corrected that it's indeed quite doable to wrap a darwin bundle without messing with bundle metadata etc., we just need to make some adjustments. I've posted https://github.com/NixOS/nixpkgs/pull/377863 as draft and will complete it some time. It's not ready yet (firefox is currently broken when any tab is opened), but it's a matter of bending the wrapping code a bit (probably some PATHs still set incorrectly pointing to wrong resources).
